hawkbit-core/src/main/java/org/eclipse/hawkbit/tenancy/configuration/TenantConfigurationProperties.java

```diff
@@ -0,0 +1,169 @@
+ *
+ */
+@ConfigurationProperties("hawkbit.server.tenant")
+public class TenantConfigurationProperties {
+
+    private final Map<String, TenantConfigurationKey> configuration = new HashMap<>();
+
+    /**
+     * @return full map of all configured tenant properties
+     */
+    public Map<String, TenantConfigurationKey> getConfiguration() {
+        return configuration;
+    }
+
+    /**
+     * @return full list of {@link TenantConfigurationKey}s
+     */
+    public Collection<TenantConfigurationKey> getConfigurationKeys() {
+        return configuration.values();
+    }
+
+    /**
+     * @param keyName
+     *            name of the TenantConfigurationKey
+     * @return the TenantConfigurationKey with the name keyName
+     */
+    public TenantConfigurationKey fromKeyName(final String keyName) {
+        return configuration.values().stream().filter(conf -> conf.getKeyName().equals(keyName)).findFirst()
+                .orElseThrow(() -> new InvalidTenantConfigurationKeyException(
+                        "The given configuration key " + keyName + " does not exist."));
+    }
+
+    /**
+     * Tenant specific configurations which can be configured for each tenant
+     * separately by means of override of the system defaults.
+     *
+     */
+    public static class TenantConfigurationKey {
+        /**
+         * Header based authentication enabled.
+         */
+        public static final String AUTHENTICATION_MODE_HEADER_ENABLED = "authentication.header.enabled";
+
+        /**
+         * Header based authentication authority name.
+         */
+        public static final String AUTHENTICATION_MODE_HEADER_AUTHORITY_NAME = "authentication.header.authority";
+        /**
+         * Target token based authentication enabled.
+         */
+        public static final String AUTHENTICATION_MODE_TARGET_SECURITY_TOKEN_ENABLED = "authentication.targettoken.enabled";
+
+        /**
+         * Gateway token based authentication enabled.
+         */
+        public static final String AUTHENTICATION_MODE_GATEWAY_SECURITY_TOKEN_ENABLED = "authentication.gatewaytoken.enabled";
+
+        /**
+         * Gateway token value.
+         */
+        public static final String AUTHENTICATION_MODE_GATEWAY_SECURITY_TOKEN_KEY = "authentication.gatewaytoken.key";
+
+        /**
+         * See system default in
+         * {@link ControllerPollProperties#getPollingTime()}.
+         */
+        public static final String POLLING_TIME_INTERVAL = "pollingTime";
+
+        /**
+         * See system default in
+         * {@link ControllerPollProperties#getPollingOverdueTime()}.
+         */
+        public static final String POLLING_OVERDUE_TIME_INTERVAL = "pollingOverdueTime";
+
+        /**
+         * See system default {@link Download#isEnabled()}.
+         */
+        public static final String ANONYMOUS_DOWNLOAD_MODE_ENABLED = "anonymous.download.enabled";
+
+        private String keyName;
+        private String defaultValue;
```

Comment:
Maybe it makes sense to initialize the `defaultValue` as empty String. Because the default class and validator is String based. Then I could just configure the configuration for the `keyName` and it will work without problem with `null` value as default, right?